### PR TITLE
[backport] bitcoin-cli: Use symbolic constants

### PR DESCRIPTION
**Pull request by kyuupichan**
This needs to be on top of #337 (Qt Segfault)

Core PRs are bitcoin#8540, bitcoin#9067 and bitcoin#9120

### CODE REVIEW

File: src/qt/bitcoin.cpp [hunk 1]
```diff
@@ -580,27 +580,28 @@ int main(int argc, char *argv[])
...
         help.showOrPrint();
-        return 1;
+        return EXIT_SUCCESS;
     }
 
     /// 5. Now that settings and translations are available, ask user for data directory
     // User language is set up: pick a data directory
-    Intro::pickDataDirectory();
+    if (!Intro::pickDataDirectory())
+        return EXIT_SUCCESS;
```
Comment (gandrewstone) [inline src/qt/bitcoin.cpp]: why would we exit with a success if we can't get the data directory?

File: src/qt/bitcoin.cpp [hunk 2]
```diff
@@ -636,7 +637,7 @@ int main(int argc, char *argv[])
     // - Do this after creating app and setting up translations, so errors are
     // translated properly.
     if (PaymentServer::ipcSendCommandLine())
-        exit(0);
+        exit(EXIT_SUCCESS);
```
Comment (gandrewstone) [inline src/qt/bitcoin.cpp:640]: why do we return everywhere else, but exit here?

Comment (kyuupichan) [reply src/qt/bitcoin.cpp:640]: Good points; thanks for the careful review Andrew.  I'll dig deeper.

Comment (kyuupichan) [reply src/qt/bitcoin.cpp:640]: Agree with both your points; updated appropriately

**Pull request merged**